components/org.wso2.bfsi.consent.management.service/src/main/java/org/wso2/bfsi/consent/management/service/util/ConsentCoreServiceUtil.java

```diff
@@ -0,0 +1,738 @@
+                log.debug(("Creating audit record for the consent update of consent ID: "
+                        + resource.getConsentID()).replaceAll("[\r\n]", ""));
+            }
+            // Create an audit record execute state change listener
+            HashMap<String, Object> consentDataMap = new HashMap<>();
+            consentDataMap.put(ConsentCoreServiceConstants.DETAILED_CONSENT_RESOURCE, resource);
+            postStateChange(connection, consentCoreDAO, resource.getConsentID(), userID,
+                    newExistingConsentStatus, previousConsentStatus,
+                    ConsentCoreServiceConstants.CREATE_EXCLUSIVE_AUTH_CONSENT_REASON, resource.getClientID(),
+                    consentDataMap);
+
+            // Extract account mapping IDs for retrieved applicable consents
+            if (log.isDebugEnabled()) {
+                log.debug(("Extracting account mapping IDs from consent ID: " +
+                        resource.getConsentID()).replaceAll("[\r\n]", ""));
+            }
+            resource.getConsentMappingResources().forEach(mappingResource ->
+                    accountMappingIDsList.add(mappingResource.getMappingID()));
+        }
+
+        // Update account mappings as inactive
+        log.debug("Deactivating account mappings");
+        consentCoreDAO.updateConsentMappingStatus(connection, accountMappingIDsList,
+                ConsentCoreServiceConstants.INACTIVE_MAPPING_STATUS);
+    }
+
+    /**
+     * Method to create an audit record in post consent state change.
+     *
+     * @param connection              Database connection
+     * @param consentCoreDAO          Consent core DAO
+     * @param consentID               Consent ID
+     * @param userID                  User ID
+     * @param newConsentStatus        New consent status
+     * @param previousConsentStatus   Previous consent status
+     * @param reason                  Reason for the status change
+     * @param clientId                Client ID
+     * @param consentDataMap          Consent data map
+     * @throws ConsentDataInsertionException If an error occurs when storing the audit record
+     * @throws ConsentManagementException    Consent management exception
+     */
+    public static void postStateChange(Connection connection, ConsentCoreDAO consentCoreDAO, String consentID,
+                                       String userID, String newConsentStatus, String previousConsentStatus,
+                                       String reason, String clientId, Map<String, Object> consentDataMap)
+            throws ConsentDataInsertionException, ConsentManagementException {
+
+        createAuditRecord(connection, consentCoreDAO, consentID, userID, newConsentStatus, previousConsentStatus,
+                reason);
+        // TODO:
+//        ConsentStateChangeListenerImpl.getInstance().onStateChange(consentID, userID, newConsentStatus,
+//                previousConsentStatus, reason, clientId, consentDataMap);
```

Comment:
Let's remove the commented lines if not required. Else let's add a proper todo comment

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/3/commits/08d281c66fc90f979dcae36a8c8baedf99ab9aa9